batch2/batch/driver/instance_pool.py

```diff
@@ -84,8 +85,7 @@ def adjust_for_remove_instance(self, instance):
         self.instances_by_last_updated.remove(instance)
         if instance.state in ('pending', 'active'):
             self.live_free_cores_mcpu -= instance.free_cores_mcpu
-        if instance.state == 'active':
-            self.active_instances_by_free_cores.remove(instance)
+        self.healthy_instances_by_free_cores.remove(instance)
```

Comment:
Is the instance guaranteed to be in the `healthy_instances_by_free_cores` when calling this function? I think this will throw an exception if it is not.

Reply:
Oops, yes, this change was totally wrong.  Fixed.  Thanks!